### PR TITLE
ICM-3662

### PR DESCRIPTION
**Pull request by anayaian**

### CODE REVIEW

File: CommerceApiSDK/Services/CategoryService.cs [hunk 1]
```diff
@@ -62,43 +65,51 @@ public async Task<List<Category>> GetCategoryList(CategoryQueryParameters parame
...
                 string url = CommerceAPIConstants.CategoryUrl;
 
                 url += parameters?.ToQueryString();
 
-                CategoryResult allCategories = await GetAsyncWithCachedResponse<CategoryResult>(
+                var response = await GetAsyncWithCachedResponse<CategoryResult>(
                     url
                 );
+                CategoryResult allCategories = response.Model;
                 List<Category> flattedCategories = FlattenCategoryTree(allCategories.Categories);
```
Comment (e128) [inline CommerceApiSDK/Services/CategoryService.cs:95]: "flattened" not "flatted"

File: CommerceApiSDK/Services/CategoryService.cs [hunk 2]
```diff
@@ -62,43 +65,51 @@ public async Task<List<Category>> GetCategoryList(CategoryQueryParameters parame
...
                 List<Category> featuredCategories = flattedCategories
```
Comment (e128) [inline CommerceApiSDK/Services/CategoryService.cs:96]: can you do without the intermediate List created on the above line?

File: CommerceApiSDK/Services/ServiceBase.cs [hunk 3]
```diff
@@ -696,15 +724,59 @@ protected async Task ClearOnlineCacheForObjects<T>()
             await this.CacheService.OnlineCache.InvalidateAllObjects<T>();
         }
 
-        private async Task LogException(HttpResponseMessage httpResponseMessage, string url, string content = null)
+        private async Task<Exception> LogException(HttpResponseMessage httpResponseMessage, string url, string content = null)
         {
             var exception = $"{GetType().FullName}\n{url}\n";
```
Comment (e128) [inline CommerceApiSDK/Services/ServiceBase.cs:729]: should use Environment.NewLine instead of '\n'

Comment (e128) [inline CommerceApiSDK/Services/ServiceBase.cs:729]: This should be a StringBuilder with proper AppendLine method calls

File: CommerceApiSDK/Services/VmiLocationsService.cs [hunk 4]
```diff
@@ -510,41 +517,58 @@ private void FixProduct(Product product)
...
 
-                    foreach (VmiBinModel item in response.VmiBins)
+                    /// TODO: Individual product responses should also be
+                    /// wrapped in ServiceResponse
+                    /// skipping it for now
+                    foreach (VmiBinModel item in response.Model?.VmiBins)
                     {
-                        GetProductResult productResult =
+                        var productResult =
                             await GetAsyncWithCachedResponse<GetProductResult>(
```
Comment (e128) [inline CommerceApiSDK/Services/VmiLocationsService.cs:535]: possible performance problem (async in a loop)

**Review by e128 — APPROVED** [review]

File: CommerceApiSDK/Services/CategoryService.cs (hunk 2 above)
Comment (anayaian) [reply CommerceApiSDK/Services/CategoryService.cs]: Check now.

File: CommerceApiSDK/Services/ServiceBase.cs (hunk 3 above)
Comment (anayaian) [reply CommerceApiSDK/Services/ServiceBase.cs]: Check now.

Comment (anayaian) [reply CommerceApiSDK/Services/ServiceBase.cs]: Check now.

**Review by e128 — APPROVED** [review]

**Review by Mosabbir-Khan — APPROVED** [review]

**Review by Mosabbir-Khan — APPROVED** [review]

**Review by tawhidjoarder — APPROVED** [review]

**Review by anisuroptimizely — APPROVED** [review]

**Pull request merged**